### PR TITLE
Add missing events for admin-related methods

### LINKED ISSUE #1173: Missing events for admin-related methods
There seems to be no event for `setContractAdmin`, `setAccessConfig`. 
Is there any particular reason you didn't add events for these methods? Because I think this information is very important for everyone who use that contract. If not, I wanna add some events for them.

### PR DESCRIPTION
**Pull request by tkxkd0159**
closes #1173

### CODE REVIEW

**Review by alpe — COMMENTED** [review]
Thanks a lot for your PR! 👏 
I had started some work yesterday already to speed this up but wasn't ready to push. You may want to 👀 at #1179 for inspiration or you review mine and we close this.

I would ask for tests in your PR. The coverage of setAdmin was not great before. I like to improve the coverage while working in an area. This helps with regression

**Comment by tkxkd0159** [comment]
> Thanks a lot for your PR! clap I had started some work yesterday already to speed this up but wasn't ready to push. You may want to eyes at #1179 for inspiration or you review mine and we close this.
Thank you for your comment @alpe  and i am fine whichever option you choose. If you wanna close this PR, I will review yours if it need it or I will add tests for this PR If this PR continue.

File: x/wasm/keeper/keeper.go [hunk 1]
```diff
@@ -639,6 +639,20 @@ func (k Keeper) setContractAdmin(ctx sdk.Context, contractAddress, caller, newAd
...
+
+	if newAdmin != nil {
+		ctx.EventManager().EmitEvent(sdk.NewEvent(
+			types.EventTypeUpdateContractAdmin,
+			sdk.NewAttribute(types.AttributeKeyContractAddr, contractAddress.String()),
+			sdk.NewAttribute(types.AttributeKeyAdmin, newAdmin.String()),
+		))
+	} else {
+		ctx.EventManager().EmitEvent(sdk.NewEvent(
+			types.EventTypeClearContractAdmin,
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply x/wasm/keeper/keeper.go:651]: I like your option as well. I just thought about this and I decided to distinguish those events like this for the view a message as a unit of action.

File: x/wasm/keeper/keeper.go [hunk 2]
```diff
@@ -976,6 +990,14 @@ func (k Keeper) setAccessConfig(ctx sdk.Context, codeID uint64, caller sdk.AccAd
 
 	info.InstantiateConfig = newConfig
 	k.storeCodeInfo(ctx, codeID, *info)
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
+		types.EventTypeSetAccessConfig,
+		sdk.NewAttribute(types.AttributeKeyCodeID, strconv.FormatUint(codeID, 10)),
+		sdk.NewAttribute(types.AttributeKeyCodePermission, newConfig.Permission.String()),
+		sdk.NewAttribute(types.AttributeKeyAuthorizedAddresses, strings.Join(newConfig.Addresses, ", ")),
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply x/wasm/keeper/keeper.go:998]: >The authorized addresses are not always set. I would prefer to set the attribute only for certain permissions where a value would be expected.

I think it's fine to add `addresses` attributes in schema perspective. The `addresses` field will just be left empty, and the client can process the event data appropriately according to the `CodePermission`.

**Comment by alpe** [comment]
I will close this PR now in favour of #1179 just to speed up the process for v0.31 since you had no strong opinion about the implementation.
I appreciate your contribution very much and I am sorry for the duplicate work. Please keep adding issues and PRs to improve the project 🌻

**Pull request closed without merging**